tests/datasets/test_dataset.py

```diff
@@ -45,18 +45,31 @@ def clear_datasets():
         pytest.param("", id="empty"),
         pytest.param("\n\t", id="whitespace"),
         pytest.param("a" * 3001, id="too_long"),
-        pytest.param("airflow:" * 3001, id="reserved_scheme"),
-        pytest.param("😊" * 3001, id="non-ascii"),
+        pytest.param("airflow://xcom/dag/task", id="reserved_scheme"),
+        pytest.param("😊", id="non-ascii"),
+        pytest.param("ftp://user@localhost/foo.txt", id="has-auth"),
```

Comment:
Is this (user info) the only breaking change for core? I know people shouldn't have done it, but I'm not sure we should blow up if they have.

Reply:
I don’t want this to be stored into the database and affect URI identity, but silently stripping it also feels a bit weird. Is it still comparably better to you?

Reply:
Cool, I’ll put this as a warning (DagWarning) instead

Reply:
#37699